packages/build/tests/time/tests.js

```diff
@@ -2,10 +2,14 @@
 
 const { version } = require('process')
 
+const { intercept, cleanAll } = require('@netlify/nock-udp')
 const test = require('ava')
 
 const { runFixture } = require('../helpers/main')
-const { startUdpServer } = require('../helpers/udp_server')
+
+test.after(() => {
+  cleanAll()
```

Comment:
Instead of directly modifying `Socket.prototype.send()`, would [`sinon.stub(Socket.prototype, 'send')`](https://sinonjs.org/releases/latest/stubs/) followed by `stub.reset()` work?

Reply:
I would need to test it, however I'm wondering what would be the benefit of it? We would still be stubbing a global prop meaning other test executions under the same process would be affected by it no? 🤔 and we don't really have a clear way to predict what will be the call pattern to `Socket.prototype.send` else, we could try to rely on mechanisms like `nth` call or, only on the first call, something like that.

I guess an alternative to this approach would be to stub `statsdClient.distribution` call. We wouldn't need the mocking module nor a lot of this extra complexity. The obvious down side being we wouldn't be validating our `statsd` payloads accurately.

Reply:
Yes, you're correct, `sinon` does not make sense here :+1: